src/pages/BasketPage/components/BasketPageContent/index.js

```diff
@@ -1,141 +1,221 @@
-  } else if (
-    useGroceres.isLoading ||
-    useGroceriesCategories.isLoading ||
-    useProducts.isLoading ||
-    useProductsCategories.isLoading ||
-    dishes.isLoading ||
-    dishesCategories.isLoading ||
-    savedList.isLoading ||
-    useSavedListsCategories.isLoading ||
-    useBasketGroceres.isLoading ||
-    useBasketProducts.isLoading ||
-    useBasketDishes.isLoading ||
-    useBasketSavedLists.isLoading
-  ) {
-    return 'Loading date...'
-  } else if (
-    useGroceres.isSuccess &&
-    useGroceriesCategories.isSuccess &&
-    useProducts.isSuccess &&
-    useProductsCategories.isSuccess &&
-    dishes.isSuccess &&
-    dishesCategories.isSuccess &&
-    savedList.isSuccess &&
-    useSavedListsCategories.isSuccess &&
-    useBasketGroceres.isSuccess &&
-    useBasketProducts.isSuccess &&
-    useBasketDishes.isSuccess &&
-    useBasketSavedLists.isSuccess
-  ) {
-    const basketEmpty =
-      useBasketGroceres.data.length === 0 &&
-      useBasketProducts.data.length === 0 &&
-      useBasketDishes.data.length === 0 &&
-      useBasketSavedLists.data.length === 0
-
-    const messege = <Styles.Message>Your basket is empty!</Styles.Message>
-
-    return basketEmpty ? (
-      messege
-    ) : editMode ? (
-      <div>
-        <EditModeList
-          parentsTitle="Groceries"
-          itemsList={useBasketGroceres.data}
-          filtredCategories={useGroceriesCategories.data}
-        />
-        <EditModeList
-          parentsTitle="Products"
-          itemsList={useBasketProducts.data}
-          filtredCategories={useProductsCategories.data}
-        />
-        <EditModeList
-          parentsTitle="Dishes"
-          itemsList={useBasketDishes.data}
-          filtredCategories={dishesCategories.data}
-        />
-        <EditModeList
-          parentsTitle="SavedList"
-          itemsList={useBasketSavedLists.data}
-          filtredCategories={useSavedListsCategories.data}
-        />
-        <Switch>
-          <Route path="/basket/createBasketHistory">
-            <Modal>
-              <CreateHistoryModalContent />
-            </Modal>
-          </Route>
-        </Switch>
-      </div>
-    ) : (
-      <div>
-        <ListModeList
-          parentsTitle="Groceries"
-          itemsList={useBasketGroceres.data}
-          filtredCategories={useGroceriesCategories.data}
-        />
-        <ListModeList
-          parentsTitle="Products"
-          itemsList={useBasketProducts.data}
-          filtredCategories={useProductsCategories.data}
-        />
-        <ListModeList
-          parentsTitle="Dishes"
-          itemsList={useBasketDishes.data}
-          filtredCategories={dishesCategories.data}
-        />
-        <ListModeList
-          parentsTitle="SavedList"
-          itemsList={useBasketSavedLists.data}
-          filtredCategories={useSavedListsCategories.data}
-        />
-      </div>
-    )
+  const groceriesCategoriesData = useSelector(selectGroceriesCategoriesData)
```

Comment:
jak się robi tak dużo podobnych bloków w komponencie, a poniżej powielasz też to EditModeList, warto to wydzielić na osobne komponenty